lite/performance_test.go

```diff
@@ -115,3 +116,112 @@ func benchmarkCertifyCommit(b *testing.B, keys lite.ValKeys) {
 	}
 
 }
+
+type algo bool
+
+const (
+	linearSearch = true
+	binarySearch = false
+)
+
+var (
+	fcs5, h5       = genFullCommits(nil, nil, 5)
+	fcs50, h50     = genFullCommits(fcs5, h5, 50)
+	fcs100, h100   = genFullCommits(fcs50, h50, 100)
+	fcs500, h500   = genFullCommits(fcs100, h100, 500)
+	fcs1000, h1000 = genFullCommits(fcs500, h500, 1000)
+)
+
+func BenchmarkMemStoreProviderGetByHeightLinearSearch5(b *testing.B) {
```

Comment:
I'd rather want to see test instead of benchmark. It's clean for any dev that binary search is faster than linear search

Reply:
Not really though, which is what prompted this change. Please see the discussion for this issue where it is explained when to use the binary search and in fact even the benchmarks illustrate that binary search starts being useful when are at say 50 items but for the common case, linear search will beat it.

Reply:
@melekes some tests already exist here https://github.com/tendermint/tendermint/blob/edbcd8d684450c182cf8e129cf19a704b422454e/lite/provider_test.go

Reply:
cool. still would be nice to have a test